syntax = "proto3";

package org.couchers.jail;

import "google/protobuf/empty.proto";

service Jail {
  // This is a secure service but allows jailed users access.

  /*
  Jail API

  If the user hasn't completed necessary steps needed to use the app, they are
  only allowed to hit the Jail API.

  E.g. if signing up is incomplete, or they haven't agreed to TOS, etc.

  These API functions are available only to signed-in users, but while other
  APIs can't be used while Jailed, these can.
  */
  rpc JailInfo(google.protobuf.Empty) returns (JailInfoRes) {
    // Get reason why the user is in jail, e.g. hasn't completed all sign-up
    // steps, or needs to read something, etc
  }

  rpc AcceptTOS(AcceptTOSReq) returns (JailInfoRes) {
    // Accept to TOS
  }

  rpc GetLocation(google.protobuf.Empty) returns (GetLocationRes) {
    // Get whether the user has added a location or not
  }

  rpc SetLocation(SetLocationReq) returns (GetLocationRes) {
    // Set the user's location
  }
}

message JailInfoRes {
  bool jailed = 1;

  bool has_not_accepted_tos = 2;
}

message AcceptTOSReq {
  bool accept = 1;
}

message GetLocationRes {
  bool has_location = 1;
}

message SetLocationReq {
  // todo
}
